crates/uv-resolver/src/lock.rs

```diff
@@ -634,6 +656,7 @@ impl SourceKind {
             SourceKind::Direct(_) => "direct",
             SourceKind::Path => "path",
             SourceKind::Directory => "directory",
+            SourceKind::Editable => "editable",
```

Comment:
Alternatively, we can reuse `SourceKind::Directory` and encode the editable somewhere else... like on the URL?

Reply:
I think a new source kind makes sense to me. Not 100% sure though.